questions-3rd-ed.md

```diff
@@ -0,0 +1,221 @@
+# Week 1
+
+## Chapter 1: Introduction
+
+- What were the major problems of Digital Currencies before Bitcoin? How did Bitcoin solve them?
+- What are the major components of the Bitcoin System? What does it mean by "consensus rules"?
+- What are the different types of Bitcoin Wallets? Which one among them is most and least secured?
+- What are the different types of Bitcoin Nodes? What are the tradeoffs between them? When is it applicable to use a "light node"?
+- Do you think it would be better if bitcoin transactions were reversible? What are the potential benefits and drawbacks?
+- Do you believe Bitcoin needs to compete with Visa/MasterCard to succeed? Why or why not?
+
+
+## Chapter 2: How Bitcoin Works
+
+- What is transaction fee? How is it calculated? Whats the effect of transaction fee on transaction confirmation?
+- What are the two main components of a Transaction? What does each component include?
+- What are change addresses? When are change addresses required? What happens if the user (or wallet sofwtware) forgets to include the change address in a transaction? When will we not require a change address?
+- What does coinselection mean? Can you name few popular coinselection algorithms? Is it better to manually select coins or let the wallet software automatically handle it?
+- How is Proof of Work mining similar to "needle in haystack" problem? What happens when two blocks are mined by different miners at the same height, how does the network decide which is the "right" block?
+- What does it mean by a transaction to be "confirmed"? How many confirmations is usually acceptable? Is it okay to accept unconfirmed transactions? Why or why not?
+
+
+--------------
+
+
+# Week 2
+
+## Chapter 3: Bitcoin Core
+
+- Why is Bitcoin Core called a "reference implementation"? What other implementation of Bitcoin are out there? Is it preferable to have many implementation of Bitcoin?
+
+- What are BIPs? What are their role in the Bitcoin development process?
+
+
+## Chapter 4: Keys and Addresses
+
+### Elliptic Curves
+
+- Describe the functions of ‘private keys’, ‘public keys’, ‘Bitcoin addresses’, and ‘digital signatures’. What is the mathematical relationship between them and how are they used in Bitcoin?
+
+- What is the discrete logarithm problem, and why is it considered hard to solve?
+
+- What is asymmetry in cryptography, and why are one-way functions useful? Identify a one-way cryptographic function used in deriving public keys from private keys.
```

Comment:
"identify the one-way problem"? There is only one right for priv > pub calculation?

Reply:
yes